server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/credential/PublicKeyMirrorCredential.java

```diff
@@ -52,47 +59,30 @@ public PublicKeyMirrorCredential(@JsonProperty("id") @Nullable String id,
 
         super(id, hostnamePatterns);
 
-        this.username = requireNonEmpty(username, "username");
-
-        requireNonEmpty(publicKey, "publicKey");
-        requireNonEmpty(privateKey, "privateKey");
-        this.publicKey = requireNonEmpty(publicKey, "publicKey").getBytes(StandardCharsets.UTF_8);
-        this.privateKey = requireNonEmpty(privateKey, "privateKey").getBytes(StandardCharsets.UTF_8);
-
-        this.passphrase = decodeBase64OrUtf8(passphrase, "passphrase");
-    }
-
-    public PublicKeyMirrorCredential(@Nullable String id,
-                                     @Nullable Iterable<Pattern> hostnamePatterns,
-                                     String username, byte[] publicKey, byte[] privateKey,
-                                     @Nullable byte[] passphrase) {
-        super(id, hostnamePatterns);
-
         this.username = requireNonEmpty(username, "username");
         this.publicKey = requireNonEmpty(publicKey, "publicKey");
-        this.privateKey = requireNonEmpty(privateKey, "privateKey");
-        this.passphrase = passphrase;
+        requireNonEmpty(privateKey, "privateKey");
+        // privateKey is converted into a list of Strings that will be used as an input of
+        // KeyPairResourceLoader.loadKeyPairs(...)
+        this.privateKey = ImmutableList.copyOf(NEWLINE_SPLITTER.splitToList(privateKey));
```

Comment:
Note: This conflicts with https://github.com/line/centraldogma/pull/838/files#diff-dc2e95ca4937a83c31177968549f54cde0f5d691aaef8e0503b6d5249f9af8b0R91 which serializes the original private key string as a JSON field for responses to a mirroring REST API. Let me fix this code later to normalize and transform the value on the call site.

Reply:
If so, we can probably add another field for that. Anyway, sorry about that making you do the job twice. 😆

Reply:
No worries. I just wanted to share the work will change in the future. :-)